src/math/CurveFitting.cc

```diff
@@ -304,31 +269,25 @@ namespace pfs{ namespace drp{ namespace stella{ namespace math{
           if (B_HaveMeasureErrors)
             V_MeasureErrors.push_back((*P_D_A1_MeasureErrors)[i_pos]);
           I_A1_OrigPos.push_back(i_pos);
-
-          I_DataValues_New++;
         }
         else{
           P_I_A1_Rejected->push_back(i_pos);
           LOGLS_DEBUG(_log, "Rejecting D_A1_X_In(" << i_pos << ") = " << D_A1_X_In[i_pos]);
-          I_NReject++;
           ++(*P_I_NRejected);
         }
       }
       if (B_HaveMeasureErrors){
         P_D_A1_MeasureErrorsTemp.reset(new ndarray::Array<T, 1, 1>(ndarray::external(V_MeasureErrors.data(),
                                                                                      ndarray::makeVector(int(V_MeasureErrors.size())),
                                                                                      ndarray::makeVector(1))));
-        LOGLS_DEBUG(_log, "D_A1_X.size() = " << D_A1_X.size());
-        LOGLS_DEBUG(_log, "V_MeasureErrors.size() = " << V_MeasureErrors.size());
-        LOGLS_DEBUG(_log, "P_D_A1_MeasureErrorsTemp.getShape()[0] = " << P_D_A1_MeasureErrorsTemp->getShape()[0]);
       }
 
       B_Run = false;
       if (*P_I_NRejected != I_NRejected_Old)
         B_Run = true;
       else{
         for (int i_pos=0; i_pos < *P_I_NRejected; i_pos++){
-          if (std::fabs((*P_I_A1_Rejected)[i_pos] - I_A1_Rejected_Old[i_pos]) > 0.0001)
+          if (((*P_I_A1_Rejected)[i_pos] - I_A1_Rejected_Old[i_pos]) != 0)
```

Comment:
As above. I actually changed it again to `if ((*P_I_A1_Rejected)[i_pos] != I_A1_Rejected_Old[i_pos])`. As both variables can only have integer values the new behavior must be the same as the old one, and all tests are passing, including the ones where we check the rejected values of PolyFit.